### PR TITLE
add batch presigned api

### PR DESCRIPTION
**Pull request by wzkun**
Our project need batch PresignedGetObject and PresignedPutObject api, but minio-go has not provide those apis, so I change the api-presigned.go file.

### CODE REVIEW

File: api-presigned.go [hunk 1]
```diff
@@ -92,6 +93,74 @@ func (c Client) PresignedPutObject(bucketName string, objectName string, expires
...
+// upto 7days or a minimum of 1sec. Additionally you can override
+// a set of response headers using the query parameters.
+func (c Client) PresignedHeadObjects(bucketName string, objectNames []string, expires time.Duration, reqParams url.Values) (u *url.URL, err error) {
+	return c.presignURLs("HEAD", bucketName, objectNames, expires, reqParams)
+}
+
+// PresignedPutObjects - Returns a presigned URL map to upload  objects
+// without credentials. URL can have a maximum expiry of upto 7days
+// or a minimum of 1sec.
+func (c Client) PresignedPutObjects(bucketName string, objectNames []string, expires time.Duration) (map[string][]*url.URL, err error) {
```
[the remark this answers is not in the document]
Comment (wzkun) [reply api-presigned.go]: OK, I will change the error.

**Comment by wzkun** [comment]
Because we need to put a few objects sometimes.
And mobile or web gets object through urls.

**Comment by harshavardhana** [comment]
Yes but you should do this outside. This API is not meaningful IMO - APIs need to be primitives. The API proposed is not a primitive. 

We shouldn't move application logic into minio-go.

**Comment by wzkun** [comment]
OK,I know that. But can minio-go provide batch PresignedPutObject and PresignedGetObject api ?

**Comment by harshavardhana** [comment]
That is exactly what I meant batch API is not an expected behavior. It should be done outside in the application.

PreSign is a Primitive which requires a bucket and an object. Look at how `mc` an application generates presigned URLs using the same API for many objects. By simply iterating over ListObjects response.

**Comment by harshavardhana** [comment]
Closing this as not required

**Pull request closed without merging**